packages/app-mobile/components/screens/ConfigScreen/plugins/PluginStates.tsx

```diff
@@ -35,6 +37,42 @@ export const getSearchText = () => {
 	return searchText;
 };
 
+const logger = Logger.create('PluginStates');
+
+// Loaded plugins: All plugins with available manifests.
+const useLoadedPluginIds = (pluginSettings: SerializedPluginSettings) => {
+	const allPluginIds = useMemo(() => {
+		return Object.keys(
+			PluginService.instance().unserializePluginSettings(pluginSettings),
+		);
+	}, [pluginSettings]);
+
+	const [pluginReloadCounter, setPluginReloadCounter] = useState(0);
+	const loadedPluginIds = useMemo(() => {
+		if (pluginReloadCounter > 0) {
+			logger.debug(`Not all plugins were loaded in the last render. Re-loading (try ${pluginReloadCounter})`);
+		}
+
+		const pluginService = PluginService.instance();
+		return allPluginIds.filter(id => !!pluginService.plugins[id]);
+	}, [allPluginIds, pluginReloadCounter]);
+	const hasLoadingPlugins = loadedPluginIds.length !== allPluginIds.length;
+
+	// Force a re-render if not all plugins have available metadata. This can happen
+	// if plugins are still loading.
+	const pluginReloadCounterRef = useRef(0);
+	pluginReloadCounterRef.current = pluginReloadCounter;
+	const timeoutRef = useRef(null);
+	if (hasLoadingPlugins && !timeoutRef.current) {
+		timeoutRef.current = shim.setTimeout(() => {
+			timeoutRef.current = null;
+			setPluginReloadCounter(pluginReloadCounterRef.current + 1);
+		}, 1000);
+	}
```

Comment:
As alternatives to this, I've considered:
1. Adding the list of loaded plugin IDs to the existing plugin Redux state.
    - Currently, [the plugin state](https://github.com/laurent22/joplin/blob/c5dfa4c0555ecc8e3c6bb7b01e4b89ffe6edc2a5/packages/lib/services/plugins/reducer.ts#L63) seems to only include **enabled** plugins, while this screen must show both enabled and disabled plugins.
    - **Downside**: This would involve making `PluginStates` a connected component. This makes testing somewhat more complicated (though still possible).
2. Having the `PluginService` emit an event when a plugin is loaded. An event listener could be added to `useLoadedPlugins`.
    - **Downside**: Requires more changes to `PluginService`.